src/Files.App/BaseLayout.cs

```diff
@@ -647,7 +647,11 @@ private void AddShellItemsToMenu(List<ContextMenuFlyoutItemViewModel> shellMenuI
 		{
 			var openWithSubItems = ItemModelListToContextFlyoutHelper.GetMenuFlyoutItemsFromModel(ShellContextmenuHelper.GetOpenWithItems(shellMenuItems));
 			var mainShellMenuItems = shellMenuItems.RemoveFrom(!UserSettingsService.AppearanceSettingsService.MoveShellExtensionsToSubMenu ? int.MaxValue : shiftPressed ? 6 : 0);
-			var overflowShellMenuItems = shellMenuItems.Except(mainShellMenuItems).ToList();
+			var overflowShellMenuItemsUnfiltered = shellMenuItems.Except(mainShellMenuItems).ToList();
+			var overflowShellMenuItems = overflowShellMenuItemsUnfiltered.Where(
+				(x, i) => (x.ItemType == ItemType.Separator && 
+				overflowShellMenuItemsUnfiltered[i + 1 <= overflowShellMenuItemsUnfiltered.Count ? i + 1 : i].ItemType == ItemType.Separator)
```

Comment:
Won't this throw an `IndexOutOfRangeException` when `i + 1 == overflowShellMenuItemsUnfiltered.Count`?

Reply:
Yes it will, I'm just blind